govc/vcsa/shutdown/poweroff.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shutdown
+
+import (
+	"context"
+	"flag"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/appliance/shutdown"
+)
+
+type powerOff struct {
+	*flags.ClientFlag
+
+	reason string
+	delay  int
+}
+
+func init() {
+	cli.Register("vcsa.shutdown.poweroff", &powerOff{})
+}
+
+func (cmd *powerOff) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	f.IntVar(&cmd.delay, "delay", 0, "Minutes after which Power off should be initiated.")
```

Comment:
question: should we default to "now" (`0`) or use a more safe value in all of these commands?

Reply:
changed to like this
```
var now int
f.IntVar(&cmd.delay,
	"delay",
	now,
	"Minutes after which Power off should be initiated.")
```